pyseir/cli.py

```diff
@@ -137,51 +137,138 @@ def _map_outputs(
             )
 
 
-def _run_all(
-    state=None,
+def _state_only_pipeline(
+    state,
+    states_only=False,
     run_mode=DEFAULT_RUN_MODE,
-    generate_reports=False,
+    generate_reports=True,
     output_interval_days=1,
+    output_dir=None,
+):
+    _infer_rt(state, states_only=states_only)
+    _run_mle_fits(state, states_only=states_only)
+    _run_ensembles(
+        state,
+        ensemble_kwargs=dict(
+            run_mode=run_mode,
+            generate_report=generate_reports,
+            covid_timeseries=nyt_dataset,
+        ),
+        states_only=states_only,
+    )
+    if generate_reports:
+        _generate_state_reports(state)
+    # remove outputs atm. just output at the end
+    # _map_outputs(
+    #     state,
+    #     output_interval_days,
+    #     states_only=states_only,
+    #     output_dir=output_dir,
+    #     run_mode=run_mode,
+    # )
+
+
+def _build_all_for_states(
+    states=[],
+    run_mode=DEFAULT_RUN_MODE,
+    generate_reports=True,
+    output_interval_days=4,
     skip_download=False,
-    states_only=False,
     output_dir=None,
     skip_whitelist=False,
 ):
-
+    # prepare data
     _cache_global_datasets()
-
     if not skip_download:
         cache_all_data()
-
     if not skip_whitelist:
         _generate_whitelist()
 
+    # run states in paralell
+    all_county_fips = []
+    for state in states:
+        all_county_fips += model_fitter.build_county_list(state)
+
+    # do everything for just states in paralell
+    states_only_func = partial(
+        _state_only_pipeline,
+        run_mode=run_mode,
+        generate_reports=generate_reports,
+        output_interval_days=output_interval_days,
+        states_only=True,
+        output_dir=output_dir,
+    )
+    p = Pool()
+    p.map(states_only_func, states)
+    p.close()
+    p.join()
+
+    #calculate culate inference
+    # parallelize async
+    p = Pool()
+    for state in states:
+        p.apply_async(_infer_rt, state)
+    p.close()
+    p.join()
+
+    #calculate ensemble
+    print(f"executing model for {len(all_county_fips)} counties")
+    p = Pool()
+    p.map(model_fitter._execute_model_for_fips, all_county_fips)
+    p.close()
+    p.join()
+
+    #calculate ensemble
+    print(f"running model for {len(all_county_fips)} counties")
+    p = Pool()
+    ensemble_func = partial(_run_county, ensemble_kwargs=dict(run_mode=run_mode, generate_report=generate_reports))
+    p.map(ensemble_func, all_county_fips)
+    p.close()
+    p.join()
+
+    #output it all
+    output_interval_days = int(output_interval_days)
+    _cache_global_datasets()
+
```

Comment:
There is one more step in here which is to generate the pyseir merged output.  I'd like to remove the "webui output mapper at some point since this is just a weird intermediate data model that exists for no reason now.

Reply:
don't we use webui thou?